.github/workflows/firmware.yml

```diff
@@ -5,13 +5,32 @@ on:
   push:
     tags:
       - v*
+  pull_request:
 
 jobs:
+  generate_target_matrix:
+    runs-on: ubuntu-latest
+    outputs:
+      target_json: ${{ steps.set_target.outputs.target }}
+    steps:
+    - name: Checkout
+      uses: actions/checkout@master
+    - name: Set target matrix
+      id: set_target
+      shell: bash
+      run: |
+        if [[ "${GITHUB_EVENT_NAME}" == "pull_request" ]]; then
+          target_list="[\"ipq40xx-generic\", \"ramips-mt76x8\", \"x86-64\", \"x86-generic\"]"
```

Comment:
braucht es wirklich `x86-generic`?

Reply:
I picked one mips (the one for 841s and R6120), one x86-64 and one x86 to cover at least some of the architectures + the ip40xx to get at least the FB 4040 and 6150v2 easily testable.

I have no strong feeling which ones are to be picked in the end. It will not drop below the ~2.0-2.5 hours of build time anyway.

If you want it removed, I will remove it.

Reply:
As requested, I added all targets to PRs as there is no time benefit in reducing the amount of targets.